### PR TITLE
Number of threads introduced in configure, to be changed by users. Default to 1 (no change of behaviour)

### PR DESCRIPTION
**Pull request by maceligueta**
**Description**
With this line, the users do not need to Google how to increase the number of threades when compiling in Windows. Users just need to change 1 to a custom int.

### CODE REVIEW

**Review by loumalouomega — DISMISSED** [review]

**Comment by philbucher** [comment]
are you sure this works?
I tried myself and never had a difference in build time. IIRC @roigcarlo said that it uses the parallel build with all available cores by default.
I tested [here](https://github.com/KratosMultiphysics/Kratos/runs/3782583457?check_suite_focus=true) with specifying 2 cores (using your flag) and in the CI at least there is no difference in build times (the CI machine has two cores, I just checked again)

**Comment by maceligueta** [comment]
This is weird, in several occasions I have been contacted by different people complaining about a very slow, single-core compilation in Windows. I also noticed this with the provided default script, but will check it again. Maybe there has been changes in the default script regarding this matter.
I am pretty sure that adding the flag is necessary, but will check it again to be 100% sure it works, and will report my checks here. I am not merging until then.

**Comment by roigcarlo** [comment]
:+1: From my side. 

Seems that depending on the VS version and other installed software the defaults for parallel builds may change, so in principle I don't see any drawback in adding that. Just bear in mind that now everyone is set to 1 by default. Maybe it is it better to comment that line and add it only in case is needed:

```ps1
@rem Enable this if your build is slow and you have a multi-core machine
@rem set KRATOS_PARALLEL_BUILD=/MP4

cmake -G"Visual Studio 16 2019" -H"%KRATOS_SOURCE%" -B"%KRATOS_BUILD%\%KRATOS_BUILD_TYPE%"          ^
-DUSE_EIGEN_MKL=OFF        ^
-DCMAKE_CXX_FLAGS="%KRATOS_PARALLEL_BUILD%"
```
No strong opinion on this.

For future reference:

`/MSX` controls the source parallelism **if passed to `cl.exe`**, which is the same as `/p:CL_MPcount=X` to `MSBuild.exe`. As a matter of fact, `/p:CL_MPcount=X` in `MSBuild.exe` causes `cl.exe` to be invoked with `/MSX`

Then, `/m:X` in `MSBuild.exe` controls the number of parallel projects in parallel, which means that multiple applications will not be built in parallel, even if all their dependencies are fulfilled. This means that the compilation order will allays be:
``` 
# MP:1
Core -> App1 -> App2 -> App3 -> etc...
```

The in project parallelism in this case depends on the default setting of the SO. Setting it to other than 1 will make projects compile in parallel:
``` 
# MP:2
      -> App1 -> App3 -> etc...
Core
      -> App2 -> etc...
```

**TL;DR:**

`/MSX` in `CMAKE_CXX_FLAGS` -> Enable multiple cpp to be compiled in parallel
`/p:CL_MPcount=X` in `cmake --build` -> Enable multiples cpp to be compiled in parallel
`/m:x` in  `cmake --build` -> Enable multiple projects to be compiled in parallel

Also `/m:x` and `/p:CL_MPcount=x` can be combined, but notice then that the individual projects **will use each x threads**. 

@philbucher, @maceligueta if VS decides that is better to set `/m:x` and not the source parallelism, but the projects that the student is compiling are "sequential" ( for example: Core -> Dem -> Fluid -> Swimming Dem" ) maybe they only notice parallelism in a fraction of the build. But again, I am ok with the changes.

**Comment by maceligueta** [comment]
I just checked the behavior in my Windows PC and I confirm that no presence of /MPX compiles with a single thread. Adding /MPX compiles with X threads. @ipouplana made the same check and confirmed the same point.
If we have different behaviors between systems, the solution by @roigcarlo (of adding a commented out line with the number of processes) seems good to me. Will push that. At least I will not have to email that line to everyone who finds the compilation extremely slow.

**Review by roigcarlo — APPROVED** [review]
